challenge.rb

```diff
@@ -1,26 +1,32 @@
 
 def capitalize_each_string(input)
   #implement your solution here
+  input.map {|s| s.capitalize}
 end
 
 def fetch_the_dog(input)
   #implement your solution here
+  input.find_all {|s| s.downcase == "dog"}
 end
 
 def no_dogs_allowed(input)
   #implement your solution here
+  input.reject {|s| s.downcase == "dog"}
 end
 
 def count_the_animals(input)
   #implement your solution here
+  input.count
 end
 
 def fetch_the_first_two(input)
   #implement your solution here
+  input.take(2)
 end
 
 def fetch_CD_animals(input)
   #implement your solution here
+  input.select{|s| s.downcase =~ /^[cd][a-z]+/}
```

Comment:
Looks good to me. What I'm about to say is nitpicky, but you can skip needing to downcase by adding the _i_ option to the end of the regex after the final / to tell it to be case insensitive.